test/unit-tests/GovernanceTimelock.t.sol

```diff
@@ -0,0 +1,268 @@
+
+        vm.expectRevert(Error.TX_ALREADY_EXECUTED.selector);
+        timelock.executeTransaction(1, address(42), 0, bytes("42"), eta);
+    }
+
+    /// @dev cannot execute tx with wrong hash
+    function test_execute_transaction_invalid_input() public {
+        vm.startPrank(admin);
+
+        uint256 eta = block.timestamp + timelock.delay();
+        timelock.scheduleTransaction(address(42), 0, bytes("42"));
+        skip(timelock.delay());
+
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.INVALID_TX_INPUT.selector);
+        timelock.executeTransaction(1, address(42), 0, bytes("42"), eta + 1);
+    }
+
+    /// @dev cannot execute tx that is still timelocked
+    function test_execute_transaction_timelocked() public {
+        vm.startPrank(admin);
+
+        uint256 eta = block.timestamp + timelock.delay();
+        timelock.scheduleTransaction(address(42), 0, bytes("42"));
+
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.TX_TIMELOCKED.selector);
+        timelock.executeTransaction(1, address(42), 0, bytes("42"), eta);
+    }
+
+    /// @dev cannot execute tx that has expired
+    function test_execute_transaction_expired() public {
+        vm.startPrank(admin);
+
+        uint256 eta = block.timestamp + timelock.delay();
+        timelock.scheduleTransaction(address(42), 0, bytes("42"));
+        skip(timelock.delay() + timelock.GRACE_PERIOD() + 1);
+
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.TX_EXPIRED.selector);
+        timelock.executeTransaction(1, address(42), 0, bytes("42"), eta);
+    }
+
+    /// @dev cannot execute tx with invalid value
+    function test_execute_transaction_invalid_value() public {
+        vm.startPrank(admin);
+
+        uint256 eta = block.timestamp + timelock.delay();
+        timelock.scheduleTransaction(address(42), 1, bytes("42"));
+        skip(timelock.delay());
+
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.INVALID_MSG_VALUE.selector);
+        timelock.executeTransaction(1, address(42), 1, bytes("42"), eta);
+    }
+
+    /// @dev failed to execute tx on dst chain
+    function test_execute_transaction_fails_on_dst() public {
+        vm.startPrank(admin);
+
+        uint256 eta = block.timestamp + timelock.delay();
+        // Use admin as dummy target address
+        timelock.scheduleTransaction(admin, 0, bytes("42"));
+        skip(timelock.delay());
+
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.EXECUTION_FAILS_ON_DST.selector);
+        timelock.executeTransaction(1, admin, 0, bytes("42"), eta);
+    }
+
+    /// @dev sets delay
+    function test_set_delay() public {
+        vm.startPrank(address(timelock));
+
+        uint256 oldDelay = timelock.delay();
+        vm.expectEmit(true, true, true, true, address(timelock));
+        emit DelayUpdated(oldDelay, 7 days);
+
+        timelock.setDelay(7 days);
+    }
+
+    /// @dev only timelock can set delay
```

Comment:
can we also add a test for calling the `setDelay()` fn through a scheduled transaction flow

Reply:
Done.